inspirehep/dojson/hep/fields/bd1xx.py

```diff
@@ -197,27 +168,72 @@ def authors2marc(self, key, value):
     """Main Entry-Personal Name."""
     value = force_force_list(value)
 
-    def get_value(value):
-        affiliations = [
+    def _get_ids(value):
+        ids = {
+            'i': [],
+            'j': [],
+        }
+        if value.get('ids'):
+            for _id in value.get('ids'):
+                if _id.get('type') == 'INSPIRE ID':
+                    ids['i'].append(_id.get('value'))
+                elif _id.get('type') == 'ORCID':
+                    ids['j'].append('ORCID:' + _id.get('value')[6:])
+                elif _id.get('type') == 'JACOW':
+                    ids['j'].append('JACOW-' + _id.get('value')[5:])
```

Comment:
`ids['j'].append(_id.get('value'))` should be sufficient, as the prefix is `JACoW` and legacy is case-insensitive.